components/server/src/prebuilds/prebuild-manager.ts

```diff
@@ -255,20 +254,29 @@ export class PrebuildManager {
         organizationId: string,
         pagination: {
             limit: number;
-            offset?: number;
+            offset: number;
+        },
+        filter: PrebuildFilter,
+        sort: {
+            field: string;
+            order: "DESC" | "ASC";
         },
-        filter?: PrebuildFilter,
     ): Promise<PrebuildWithStatus[]> {
         await this.auth.checkPermissionOnOrganization(userId, "read_prebuild", organizationId);
 
         const prebuiltWorkspaces = await this.workspaceDB
             .trace(ctx)
-            .findPrebuiltWorkspacesByOrganization(organizationId, pagination.offset, pagination.limit, filter);
-        const prebuildMap = new Map(prebuiltWorkspaces.map((prebuild) => [prebuild.id, prebuild]));
-        const infos = await this.workspaceDB.trace({}).findPrebuildInfos([...prebuildMap.keys()]);
+            .findPrebuiltWorkspacesByOrganization(organizationId, pagination, filter, sort);
+        const prebuildIds = prebuiltWorkspaces.map((prebuild) => prebuild.id);
+        const infos = await this.workspaceDB.trace({}).findPrebuildInfos(prebuildIds);
+        const prebuildInfosMap = new Map(infos.map((info) => [info.id, info]));
+
+        return prebuiltWorkspaces.map((prebuild) => {
+            const info = prebuildInfosMap.get(prebuild.id);
+            if (!info) {
+                throw new Error("Could not find prebuild info for prebuild " + prebuild.id);
```

Comment:
Seems a little bit dangerous, one failed case can break whole view even other prebuils are ok. How about skiping this item when notFound and print error/debug if necessary.

PS. We only throw `ApplicationError` on server